src/figuresClasses.ts

```diff
@@ -2,10 +1,115 @@
+export interface Figure {
+  color: string;
+  a?: number;
+  b?: number;
+  c?: number;
+  radius?: number;
+  width?: number;
+  height?: number;
+}
+
+export type FiguresData = {
+  shape: string;
+  color: string;
+  area: number;
+};
+
+export class Triangle implements Figure {
+  public shape = 'triangle';
+
+  public area = this.getArea();
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      color === undefined ||
+      a === undefined ||
+      b === undefined ||
+      c === undefined
+    ) {
+      throw new Error('All values must be entered');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides values must be > 0');
+    }
+
+    const sides = [a, b, c];
+
+    sides.splice(sides.indexOf(Math.max(a, b, c), 1));
+
+    if (Math.max(a, b, c) >= sides[0] + sides[1]) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const s = (a + b + c) / 2;
+
+    return +(
+      Math.floor(Math.sqrt(s * (s - a) * (s - b) * (s - c)) * 100) / 100
+    );
```

Comment:
[CODE STYLE] - The `getArea` function should not accept any parameters. Instead, it should use the values available as object fields, which you are already doing. Therefore, the explicit parameter passing in the constructor is unnecessary.